### PR TITLE
Adding/fixing Alt+Space handling

### LINKED ISSUE #7125: Add an option to disable the Windows menu on alt+space in Terminal
# Description of the new feature/enhancement

This is in some sense the opposite feature from #5833: I'm requesting an option to _not_ open the Windows menu when `alt+space` is pressed.

I have until now been working with WSL from the Powershell terminal, where the `alt+space` shortcut to open the Windows menu is disabled. This is convenient since I have `alt+space` mapped as a shortcut in several WSL applications (such as screen/tmux prefixes), but it no longer works when switching to Terminal.

The usual way of unbinding keys doesn't seem to do anything here:
```
{ "command": "unbound", "keys": "alt+space" }
```
It would be nice to have some way of unbinding this in Terminal.

### PR DESCRIPTION
**Pull request by FWest98**
## Summary of the Pull Request
This PR implements/solves #7125. Concretely: two requests regarding alt+space were posted there:
1. Disabling the alt+space menu when the keychord explicitly unbound - and forwarding the keystroke to the terminal
2. Disabling the alt+space menu when the keychord is bound to an action

 
## References
Not that I know

## PR Checklist
* [x] Closes #7125
* [x] CLA signed.
* [x] Tests added/passed
* [x] Documentation updated. N/A
* [x] Schema updated.
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan.
The issue was marked Help-Wanted. I am happy to change the implementation to better fit your (planned) architecture.

## Detailed Description of the Pull Request / Additional comments

While researching the solution, I noticed that the XAML system was always opening the system menu after Alt+Space, even when explicitly setting the event to be handled according to the documentation. The only solution I could find was to hook into the "XAML bypass" already in place for F7 KeyDown, and Alt KeyUp keystrokes. This bypass sends the keystroke to the AppHost immediately. This bypass method will "fall back" to the normal XAML routing when the keystroke is not handled.

The implemented behaviour is as follows:
- Default: same as normal; system menu is working since the bypass does not handle the keystroke
- Alt+Space explicitly unbound: bypass passes the keystroke to the terminal and marks it as handled
- Alt+Space bound to command: bypass invokes the command and marks it as handled

Concretely, added a method to the KeyBindings and ActionMap interfaces to check whether a keychord is explicitly unbound. The implementation for `_GetActionByKeyChordInternal` already distinguishes between explicitly unbound and lack of binding, however this distinction is not carried over to the public methods. I decided not to change this existing method, to avoid breaking other stuff and to make the API more explicit.

Furthermore, there were some checks against Alt+Space further down in the code, preventing this keystroke from being entered in the terminal. Since the check for this keystroke is now done at a "higher" level, I thought I could safely remove these checks as otherwise the keystroke could never be sent to the terminal itself. Please correct me if I'm wrong.

Note that when alt+space is bound to an action that opens the command pallette (such as tab search), then a second press of the key combination does still open the system menu. This is because at that point, the "bypass" is cancelled (called "not a good implementation" in #4031). I don't think this can easily be solved for now, but this is a very minor bug/inconvenience.

## Validation Steps Performed
Added tests for the new method. Performed manual checking:
* [x] Default configuration still opens system menu like normal
* [x] Binding alt+space to an action performs the action and does not show the system menu
* [x] Explicitly unbinding alt+space no longer shows the system menu and sends the keystroke to the terminal. I was unable to run the debug tap (it crashed my instance - same thing happening on preview and release builds) to check for sure, but behaviour was identical to native linux terminals.

### CODE REVIEW

**Review by carlos-zamora — APPROVED** [review]
Looks good to me! Thanks!

I assume it works fine, but mind testing this with PowerToys Run as well? I imagine they intercept it and handle it before we even get the chance to see it.

File: src/cascadia/TerminalControl/TermControl.cpp [hunk 1]
```diff
@@ -813,6 +813,37 @@ namespace winrt::Microsoft::Terminal::Control::implementation
...
+                // If it is explicitly unbound, we send it to the terminal
+                if (!handled && bindings.IsKeyChordExplicitlyUnbound(kc))
+                {
+                    (void)_TrySendKeyEvent(VK_SPACE, scanCode, modifiers, true);
+                    // If sent to the terminal, we mark it handled so that it does not
+                    // get processed by the normal XAML handlers
+                    handled = true;
+                }
+            }
+        }
```
[the remark this answers is not in the document]
Comment (FWest98) [reply src/cascadia/TerminalControl/TermControl.cpp]: I did that now!

**Comment by FWest98** [comment]
@carlos-zamora Thanks for the feedback! I tested, and as suspected, PT Run makes use of some global event handler so the keystroke never arrives at the terminal.

**Review by zadjii-msft — APPROVED** [review]
Code looks good to me - thanks for doing this!

**Comment by carlos-zamora** [comment]
@msftbot merge this in 5 minutes

**Pull request merged**